### PR TITLE
refactor: Moved Nstack's appOpen() out of MainActivity

### PR DESCRIPTION
**Pull request by Tgo1014**
**Acceptance criteria:**  
- [ ] PR title includes Jira ticket reference
- [X] Follows UI patterns from [UI manifesto](https://github.com/monstar-lab-oss/ui-manifesto)
- [X] Loading/error/empty states taken into account
- [X] Status of Jira ticket updated
- [X] Removed commented out code  
- [X] Removed unnecessary logging
- [X] Tested flows in airplane mode and bad connection
- [X] Updated documentation (please add a link if other than README 😊)

### CODE REVIEW

File: app/src/main/java/com/monstarlab/features/main/MainActivityViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                nstackData = data,
+                showSplash = false
+            )
+        }
+    }
+
+    data class State(
+        val showSplash: Boolean = true,
+        val nstackData: AppOpenData? = null
+    )
```
Comment (levinzonr) [inline app/src/main/java/com/monstarlab/features/main/MainActivityViewModel.kt:43]: Lets align with other screens and move this into separate file 🙏

Comment (levinzonr) [reply app/src/main/java/com/monstarlab/features/main/MainActivityViewModel.kt:43]: I mean the State

File: app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+class SetupNstackUseCase @Inject constructor(
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) {
+
+    suspend operator fun invoke(): Result<AppOpenData> = suspendRunCatching {
+        withContext(ioDispatcher) {
+            when (val result = NStack.appOpen()) {
+                is dk.nodes.nstack.kotlin.models.Result.Success -> result.value.data
+                is dk.nodes.nstack.kotlin.models.Result.Error -> throw Exception("Failed to appOpen NStack")
```
Comment (levinzonr) [inline app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt:19]: Can we import these guys

**Review by levinzonr — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt (hunk 2 above)
Comment (Tgo1014) [reply app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt:19]: No, we have `kotlin.Result` already imported and NStack use their own custom `Result` implementation.

Comment (Tgo1014) [reply app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt:19]: I'll make a custom alias

**Review by levinzonr — APPROVED** [review]

**Pull request merged**